http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -301,7 +303,13 @@ protected void configure(SelectableChannel selectable)
         }
         HostAndPort httpProxy = config.getHttpProxy();
         if (httpProxy != null) {
-            httpClient.getProxyConfiguration().addProxy(new HttpProxy(new Address(httpProxy.getHost(), httpProxy.getPortOrDefault(8080)), config.isSecureProxy()));
+            Address proxyAddress = new Address(httpProxy.getHost(), httpProxy.getPortOrDefault(8080));
+            HttpProxy proxy = new HttpProxy(proxyAddress, config.isSecureProxy());
+            httpClient.getProxyConfiguration().addProxy(proxy);
+            if (config.getHttpProxyUser().isPresent() && config.getHttpProxyPassword().isPresent()) {
+                httpClient.getAuthenticationStore().addAuthentication(
+                        new BasicAuthentication(proxy.getURI(), ANY_REALM, config.getHttpProxyUser().get(), config.getHttpProxyPassword().get()));
```

Comment:
Can we apply them pre-emptively - So we could avoid the roundtrips ?

Reply:
httpClient.getAuthenticationStore().addAuthenticationResult(
                        new BasicAuthentication.BasicResult(
                                proxy.getURI(), 
                                HttpHeader.PROXY_AUTHORIZATION, 
                                config.getHttpProxyUser().get(), 
                                config.getHttpProxyPassword().get()));